packages/web-components/fast-foundation/src/carousel/carousel.ts

```diff
@@ -0,0 +1,411 @@
+import { attr, observable, DOM } from "@microsoft/fast-element";
+import { Tabs } from "../tabs";
+import {
+    keyCodeSpace,
+    keyCodeEnter,
+    wrapInBounds,
+    limit,
+    KeyCodes,
+} from "@microsoft/fast-web-utilities";
+import { applyMixins } from "../utilities";
+import { ARIAGlobalStatesAndProperties } from "../patterns";
+
+/**
+ * An Carousel Custom HTML Element.
+ * Implements the {@link https://www.w3.org/TR/wai-aria-1.1/#carousel | ARIA carousel }.
+ *
+ * @public
+ */
+export class Carousel extends Tabs {
+    @attr({ mode: "fromView" })
+    public autoplay: boolean = true;
+
+    @attr({ mode: "fromView" })
+    public loop: boolean = true;
+
+    @attr({ mode: "boolean" })
+    public paused: boolean = false;
+    public pausedChanged(): void {
+        if (!this.paused) {
+            this.focused = false;
+            this.notabfocus = true;
+            this.startAutoPlay();
+        } else {
+            this.notabfocus = false;
+            this.stopAutoPlay();
+        }
+        if (this.change) {
+            this.change();
+        }
+    }
+
+    @attr({ attribute: "basicpattern", mode: "boolean" })
+    public basicPattern: boolean = false;
+
+    @attr({ attribute: "autoplay-interval" })
+    public autoplayInterval: number = 6000;
+
+    @attr({ attribute: "activeslideid" })
+    public activeSlideId: string;
+    public activeSlideIdChanged(): void {
```

Comment:
Can you add tsdoc comments for all of these public properties, and ensure that any public properties *not* intended to be used by app authors are tagged as `@internal`